### DISCUSSION TITLE
How to connect with custom MongoDB collections?

### DISCUSSION THREAD

**Discussion by jahirfiquitiva** [nextauthjs/next-auth#1456] [Help]
Hello... from what I've understood from documentation, `next-auth` will automatically generate collections for users and other stuff.

In my database, I already have a users collection, and the schema is something like:
```
name: string;
username: string;
email: string;
age: number;
gender: string;
phone: string;
city: string;
country: string;
resources: {
  type: [Schema.Types.ObjectId],
},
```

Since `next-auth` won't provide all of these, how can I create users with this custom model, with the data retrieved with `next-auth`.

I mean, I'm ok with some fields being empty. But would like `next-auth` to create users in this collection with this specific model, but also keep the other collections created by `next-auth` just as they are.

I'd really appreciate any guidance and/or help on how to achieve this.
Thanks in advance!

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [x] Did not find the example project helpful

**Comment by un-focused** [comment]
I'm fairly new to this library myself so take this with a grain of salt. I'm currently migrating users from my previous structure to this structure so I had to modify the schema to account for that.

I would fork the library (make your own copy of it) & then modify the models to work accordingly. You can see the models for the adapters [here](https://github.com/nextauthjs/next-auth/tree/76df2b5e702be2084674c81e098d989c2ec39ed9/src/adapters). Once done, look where it is being called & update the callers so that you pass the relevant data to the models.

Take a look at the adapters as they are in charge of creating the users [here](https://github.com/nextauthjs/next-auth/tree/82dd6ba3e4ba481c5003c918eb1d2fa95f7d76dd/src/adapters). There is an example adapter [here](https://github.com/nextauthjs/next-auth/blob/82dd6ba3e4ba481c5003c918eb1d2fa95f7d76dd/src/adapters/example/index.js).

Hope that guides you a bit

**Reply by jahirfiquitiva** [reply]
Thanks for the information. That looks like a huge amount of work but I'll take a look into it. 👌
